src/exec.cpp

```diff
@@ -491,10 +489,6 @@ static bool exec_internal_builtin_proc(parser_t &parser, job_t *j, process_t *p,
     // make exec handle things.
     const int fg = j->get_flag(JOB_FOREGROUND);
     j->set_flag(JOB_FOREGROUND, false);
-
-    // Main loop may need to perform a blocking read from previous command's output.
```

Comment:
Not having this comment here made figuring out hangs in the job loop very difficult. Can we just a put a nice `//Note that this call may block while the builtin performs IO` or something?

Reply:
Good idea.